martin/src/bin/martin.rs

```diff
@@ -9,7 +9,7 @@ use martin::{read_config, Config, MartinResult};
 
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 
-async fn start(args: Args) -> MartinResult<Server> {
+async fn start(args: Args) -> MartinResult<impl Future<Output = MartinResult<()>>> {
```

Comment:
this looks ... scary. Could you add a comment explaining why you have future of a future here, or if we can somehow simplify this?  Basically here we have a desugared `Future<Result<Future<Result>>>>`, right?

Reply:
Exactly, the actix `Server` type was just being used as a future, and desugars to this. I note that the main function is already awkward with the `.await.unwrap().await.unwrap()`.

The only reason to do anything with the server besides awaiting it immediately would be to pause or stop it, and also to print out the listening address as we do here.

Defining a type alias for what the "Server" is would reduce the clutter in the other file, but this function can really be changed to return `MartinResult<()>`, since this is essentially the main.